frontend/src/scenes/surveys/SurveyEdit.tsx

```diff
@@ -124,19 +86,169 @@ export default function SurveyEdit(): JSX.Element {
             value: 'until_adaptive_limit',
             label: 'Collect a certain number of surveys per day, week or month',
             'data-attr': 'survey-collection-until-adaptive-limit',
-            disabledReason: surveysAdaptiveLimitsDisabledReason,
+            disabledReason: surveysRecurringScheduleAvailable
+                ? undefined
+                : 'Upgrade your plan to use an adaptive limit on survey responses',
         } as unknown as LemonRadioOption<'until_stopped' | 'until_limit' | 'until_adaptive_limit'>)
     }
 
-    useMemo(() => {
-        if (surveyUsesLimit) {
-            setDataCollectionType('until_limit')
-        } else if (surveyUsesAdaptiveLimit && adaptiveLimitFFEnabled) {
-            setDataCollectionType('until_adaptive_limit')
-        } else {
-            setDataCollectionType('until_stopped')
-        }
-    }, [surveyUsesLimit, surveyUsesAdaptiveLimit, adaptiveLimitFFEnabled, setDataCollectionType])
+    return (
+        <>
+            <div className="mt-2">
+                <h3> How long would you like to collect survey responses? </h3>
+                <LemonField.Pure>
+                    <LemonRadio
+                        value={dataCollectionType}
+                        onChange={(newValue: 'until_stopped' | 'until_limit' | 'until_adaptive_limit') => {
```

Comment:
not for this PR but i'd love to start using const instead of hard coded values (the problem is the repetition) and I love defensive programming

Reply:
i agree, for this one there was a type already, [so I updated it](https://github.com/PostHog/posthog/pull/28151/commits/6000d38b8e789d376f498cee2f9fe31fd426aaf0)